core/utils/IOUtils.py

```diff
@@ -0,0 +1,14 @@
+# THE PARSER
+# reads and returns the seq as string from the FASTA/FASTQ file
+import random
+
+from Bio import SeqIO
+from Bio.SeqRecord import SeqRecord
+
+
+def get_seqs_from_file(file_path, fmt="fasta") -> list[SeqRecord]:
+    seqs = []
+    for seq_record in SeqIO.parse(file_path, fmt):
```

Comment:
If `SeqIO.parse()` returns an iterator over strings, iterating over it and making a list isn't necessary. It probably isn't necessary to make a list at all, but if you want to, then `list(SeqIO.parse(file_path, fmt))` will make the list for you. That will be both simpler and faster.